soc/arm/microchip_mec/mec172x/soc.h

```diff
@@ -24,6 +24,224 @@
 #define __FPU_PRESENT  CONFIG_CPU_HAS_FPU
 #define __MPU_PRESENT  CONFIG_CPU_HAS_ARM_MPU
 
+#define __CM4_REV 0x0201                /*!< Core Revision r2p1 */
+
+#define __VTOR_PRESENT 1                /*!< Set to 1 if VTOR is present */
+#define __NVIC_PRIO_BITS 3              /*!< Number of Bits used for Priority Levels */
+#define __Vendor_SysTickConfig 0        /*!< 0 use default SysTick HW */
+#define __FPU_DP 0                      /*!< Set to 1 if FPU is double precision */
+#define __ICACHE_PRESENT 0              /*!< Set to 1 if I-Cache is present */
+#define __DCACHE_PRESENT 0              /*!< Set to 1 if D-Cache is present */
+#define __DTCM_PRESENT 0                /*!< Set to 1 if DTCM is present */
+
+/** @brief ARM Cortex-M4 NVIC Interrupt Numbers
+ * CM4 NVIC implements 16 internal interrupt sources. CMSIS macros use
+ * negative numbers [-15, -1]. Lower numerical value indicates higher
+ * priority.
+ * -15 = Reset Vector invoked on POR or any CPU reset.
+ * -14 = NMI
+ * -13 = Hard Fault. At POR or CPU reset all faults map to Hard Fault.
+ * -12 = Memory Management Fault. If enabled Hard Faults caused by access
+ *       violations, no address match, or MPU mismatch.
+ * -11 = Bus Fault. If enabled pre-fetch, AHB access faults.
+ * -10 = Usage Fault. If enabled Undefined instructions, illegal state
+ *       transition (Thumb -> ARM mode), unaligned, etc.
+ * -9 through -6 are not implemented (reserved).
+ * -5 System call via SVC instruction.
+ * -4 Debug Monitor.
+ * -3 not implemented (reserved).
+ * -2 PendSV for system service.
+ * -1 SysTick NVIC system timer.
+ * Numbers >= 0 are external peripheral interrupts.
+ */
+typedef enum IRQn {
```

Comment:
@jvasanth1 I believe this should be typedef enum { 
...
} IRQn_Type

See https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/ti_simplelink/cc32xx/soc.h

Reply:
updated in the latest push
[soc/arm/microchip_mec/mec172x/soc.h](https://github.com/MicrochipTech/zephyr/blob/IRQn_Type_fix/soc/arm/microchip_mec/mec172x/soc.h)